jscomp/others/dom_formData.ml

```diff
@@ -0,0 +1,44 @@
+(* Provides a way to easily construct a set of key/value pairs representing form fields and their values, which can then be easily sent using the XMLHttpRequest.send() method. It uses the same format a form would use if the encoding type were set to "multipart/form-data". *)
+
+type file
+type blob
+type entryValue
+
+let classify : entryValue -> [> `String of string | `File of file ] =
+  fun t ->
+  if Js.typeof t = "string" then `String (Obj.magic t)
+  else `File (Obj.magic t)
+
+type t
+
+external make : unit -> t = "FormData" [@@mel.new]
+external append : string -> string -> unit = "append" [@@mel.send.pipe: t]
```

Comment:
`append` also takes a `?filename`

Reply:
append can't have a filename when it's a string. `appendBlob` or `appendFile` is used and have the ?filename there